keps/sig-apps/3973-consider-terminating-pods-deployment/README.md

```diff
@@ -0,0 +1,799 @@
+  rolling out new revisions and/or scaling the deployment at any point in time.
+
+<!--
+- <test>: <link to test coverage>
+-->
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature gate disabled by default.
+- Unit, e2e, and integration tests implemented and passing.
+
+
+#### Beta
+- Feature gate enabled by default.
+- Any test that checks Deployment and Replicaset status is updated to count updates to `.status.TerminatingReplicas`.
+- `.spec.podReplacementPolicy` is nil by default and preserves the original behavior.
+- E2e and integration tests are in Testgrid and linked in the KEP.
+- add new metrics to `kube-state-metrics`
+
+
+#### GA
+- Every bug report is fixed.
+- Confirm the stability of e2e and integration tests.
+- DeploymentPodReplacementPolicy feature gate is ignored.
+
+### Upgrade / Downgrade Strategy
+
+No changes required for existing cluster to use the enhancement.
+
+### Version Skew Strategy
+
+
+This feature doesn't depend on the version for nodes.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: DeploymentPodReplacementPolicy
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+
+No, the behavior is only changed when users specify the `podReplacementPolicy` in
+the Deployment spec.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. 
+
+By disabling the feature:
+- Extra pods can appear during a deployment rollout or scaling. This can increase the number of pods
+  that need to be scheduled, and it can have an impact on the resource consumption.
+- Actors reading `.status.TerminatingReplicas` for ReplicaSet and Deployments will observe 0 pods.
```

Comment:
So if the FG is disabled, the ReplicaSet/Deployment controllers will actually set the TerminatingReplicas to 0? Or rather will not touch it at all?

[In particular the controller will certainly not touch it if rolled back to a version that doesn't support the field]